include/openenclave/bits/seal.h

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/**
+ * @file seal.h
+ *
+ * This file defines constants and structures for sealing APIs.
+ *
+ * Only TEE agnostic definitions should go to this file. SGX specific
+ * definitions should go to sgx/seal.h
+ */
+
+#ifndef _OE_BITS_SEAL_H
+#define _OE_BITS_SEAL_H
+
+#include <openenclave/bits/evidence.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+typedef struct _oe_seal_setting_header
+{
+    int name;
+    size_t size;
+} oe_seal_setting_header_t;
+
+/**
+ * TEE agnostic seal settings supported by OE SDK
+ */
+typedef enum _oe_seal_setting_name
+{
+    OE_SEAL_SETTING_POLICY,
+    OE_SEAL_SETTING_ADDITIONAL_CONTEXT,
+    OE_SEAL_SETTING_TEE_AGNOSTIC_MAX
+} oe_seal_setting_name_t;
+
+typedef struct _oe_seal_setting_policy
+{
+    const oe_seal_setting_header_t header;
+    oe_seal_policy_t policy;
+} oe_seal_setting_policy_t;
```

Comment:
I would suggest, as in the strawman proposal, that we add convenience macros for initializing each of these structs, since the setting of the type and size in the header is not something that the dev needs to do (and may do incorrectly). In that case, we may choose to have the data structures themselves be defined in a bits/seal.h, but the macros we intend for the developers to use be in the top level header. 

```c
#define OE_INIT_SEAL_SETTING_POLICY(SEAL_POLICY)           \
    {                                                      \
        .header.type = OE_SEAL_SETTING_TYPE_POLICY,        \
        .header.size = sizeof(oe_seal_setting_policy_t),   \
        .policy = SEAL_POLICY                              \
    }
```